packages/client/src/reactions/Reactions.ts

```diff
@@ -1,37 +1,76 @@
+import { assertError, failure, PromiseResult, success } from '@lens-protocol/shared-kernel';
 import { GraphQLClient } from 'graphql-request';
 
-import { Credentials } from '../authentication';
+import { Authentication } from '../authentication';
 import { LensConfig } from '../consts/config';
-import { NotAuthenticatedError } from '../consts/errors';
+import { CredentialsExpiredError, NotAuthenticatedError } from '../consts/errors';
 import { ReactionRequest } from '../graphql/types.generated';
 import { getSdk, Sdk } from './graphql/reactions.generated';
 
 export class Reactions {
+  private readonly authentication: Authentication | undefined;
   private readonly sdk: Sdk;
 
-  constructor(config: LensConfig, private readonly credentials?: Credentials) {
+  constructor(config: LensConfig, authentication?: Authentication) {
     const client = new GraphQLClient(config.environment.gqlEndpoint);
 
-    if (credentials) {
-      client.setHeader('authorization', `Bearer ${credentials.accessToken}`);
-    }
-
     this.sdk = getSdk(client);
+    this.authentication = authentication;
   }
 
-  async add(request: ReactionRequest) {
-    if (!this.credentials) {
-      throw new NotAuthenticatedError();
+  async add(
+    request: ReactionRequest,
+  ): PromiseResult<void, CredentialsExpiredError | NotAuthenticatedError | Error> {
```

Comment:
the amount of code to return PromiseResult from the action method doesn't look good to me.